File: src/main/java/shtykh/roomplanner/model/RoomsUsage.java
package shtykh.roomplanner.model;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@EqualsAndHashCode
@AllArgsConstructor
@ToString
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class RoomsUsage {

    private final RoomLevel roomLevel;
    private int roomsNumber = 0;
    private int paymentSum  = 0;

    public void add(int payment) {
        roomsNumber++;
        paymentSum += payment;
    }
}

File: src/main/kotlin/shtykh/roomplanner/model/RoomsUsage.kt
package shtykh.roomplanner.model

import com.fasterxml.jackson.annotation.JsonAutoDetect

@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
data class RoomsUsage(var roomLevel: RoomLevel, var roomsNumber: Int = 0, var paymentSum: Int = 0) {

    fun add(payment: Int) {
        roomsNumber++
        paymentSum += payment
    }
}
